pkg/operator/controller/externaldns/pod.go

```diff
@@ -178,6 +186,26 @@ func (b *externalDNSContainerBuilder) fillProviderAgnosticFields(seq int, zone s
 
 	container.Args = append(container.Args, filterArgs...)
 	container.Args = append(container.Args, args...)
+
+	//
+	// ENV
+	//
+	env := []corev1.EnvVar{}
```

Comment:
this local variable not required, instead we can use directly `container.Env` in subsequent places 
like :
    ```
if utils.EnvProxySupportedProvider(b.externalDNS) {
		if val := os.Getenv(httpProxyEnvVar); val != "" {
			container.Env = append(container.Env, corev1.EnvVar{Name: httpProxyEnvVar, Value: val})
		}
		if val := os.Getenv(httpsProxyEnvVar); val != "" {
			container.Env = append(container.Env, corev1.EnvVar{Name: httpsProxyEnvVar, Value: val})
		}
		if val := os.Getenv(noProxyEnvVar); val != "" {
			container.Env = append(container.Env, corev1.EnvVar{Name: noProxyEnvVar, Value: val})
		}
	}

```

Reply:
Yeah, just kept the same pattern as for the args. Not really needed here, will change.